### PR TITLE
intial patches to build JDK8u on XCode 11

### PR DESCRIPTION
**Pull request by gdams**

### CODE REVIEW

**Comment by gdams** [comment]
test build available here: https://ci.adoptopenjdk.net/view/work%20in%20progress/job/jdk8u-mac-x64-hotspot-notarized/6/

File: common/autoconf/flags.m4 [hunk 1]
```diff
@@ -159,6 +159,26 @@ AC_DEFUN_ONCE([FLAGS_SETUP_COMPILER_FLAGS_FOR_LIBS],
     C_FLAG_REORDER=''
     CXX_FLAG_REORDER=''
 
+    if test "x$OPENJDK_TARGET_OS" = xmacosx; then
```
Comment (karianna) [inline common/autoconf/flags.m4:162]: Add a comment block at the top of this if/elif

File: common/autoconf/flags.m4 [hunk 2]
```diff
@@ -159,6 +159,26 @@ AC_DEFUN_ONCE([FLAGS_SETUP_COMPILER_FLAGS_FOR_LIBS],
...
+      # Linking is different on MacOSX
```
Comment (karianna) [inline common/autoconf/flags.m4:163]: How is it different?  Add to the comment

File: common/autoconf/flags.m4 [hunk 3]
```diff
@@ -159,6 +159,26 @@ AC_DEFUN_ONCE([FLAGS_SETUP_COMPILER_FLAGS_FOR_LIBS],
...
+      SHARED_LIBRARY_FLAGS="-dynamiclib -compatibility_version 1.0.0 -current_version 1.0.0 $PICFLAG"
+      SET_EXECUTABLE_ORIGIN='-Xlinker -rpath -Xlinker @loader_path/.'
+      SET_SHARED_LIBRARY_ORIGIN="$SET_EXECUTABLE_ORIGIN"
+      SET_SHARED_LIBRARY_NAME='-Xlinker -install_name -Xlinker @rpath/[$]1'
+      SET_SHARED_LIBRARY_MAPFILE=''
+    else
+      # Default works for linux, might work on other platforms as well.
```
Comment (karianna) [inline common/autoconf/flags.m4:170]: What sort of platforms might it work on?

File: common/autoconf/flags.m4 [hunk 4]
```diff
@@ -348,6 +379,20 @@ AC_DEFUN_ONCE([FLAGS_SETUP_COMPILER_FLAGS_FOR_OPTIMIZATION],
         C_O_FLAG_NORM="-O2"
         C_O_FLAG_NONE="-O0"
       fi
+    elif test "x$TOOLCHAIN_TYPE" = xclang; then
+      if test "x$OPENJDK_TARGET_OS" = xmacosx; then
+        # On MacOSX we optimize for size, something
+        # we should do for all platforms?
```
Comment (karianna) [inline common/autoconf/flags.m4:385]: Add a TODO or an issue for that Q

**Review by karianna — APPROVED** [review]
Approving, the feedback is nitpicked really

**Comment by karianna** [comment]
I'll hold off merging until the test run is proven to be OK

**Comment by gdams** [comment]
closing in favour of https://github.com/AdoptOpenJDK/openjdk-jdk8u/pull/10

**Pull request closed without merging**